app/scripts/lib/createConfirmationValidationMiddleware.ts

```diff
@@ -0,0 +1,53 @@
+import { errorCodes } from 'eth-rpc-errors';
+import { isValidAddress } from 'ethereumjs-util';
+import { isValidHexAddress, JsonRpcRequest } from '@metamask/utils';
+import { JsonRpcMiddleware } from 'json-rpc-engine';
+
+import { MESSAGE_TYPE } from '../../../shared/constants/app';
+import { parseTypedDataMessage } from '../../../shared/modules/transaction.utils';
+
+import { EIP712_PRIMARY_TYPE_PERMIT } from '../../../shared/constants/transaction';
+
+/**
+ * Returns a middleware that validated incoming confirmations
+ */
+
+export default function createConfirmationValidationMiddleware(): JsonRpcMiddleware<
```

Comment:
Does `Confirmation` make sense here since at this stage they're all just JSON-RPC requests, some of which may create confirmations down the line after they've also been partially validated by the controllers?

Would this instead just be a general validation middleware?